src/Middleware/RateLimiting/src/RateLimiterOptions.cs

```diff
@@ -44,5 +36,121 @@ public PartitionedRateLimiter<HttpContext> Limiter
     /// This status code will be set before <see cref="OnRejected"/> is called, so any status code set by
     /// <see cref="OnRejected"/> will "win" over this default.
     /// </remarks>
-    public int DefaultRejectionStatusCode { get; set; } = StatusCodes.Status503ServiceUnavailable;
+    public int RejectionStatusCode { get; set; } = StatusCodes.Status503ServiceUnavailable;
+
+    /// <summary>
+    /// Adds a new rate limiting policy with the given name.
+    /// </summary>
+    /// <param name="policyName">The name to be associated with the given <see cref="RateLimiter"/>.</param>
+    /// <param name="partitioner">Method called every time an Acquire or WaitAsync call is made to figure out what rate limiter to apply to the request.</param>
+    public RateLimiterOptions AddPolicy<TPartitionKey>(string policyName, Func<HttpContext, RateLimitPartition<TPartitionKey>> partitioner)
+    {
+        if (policyName == null)
+        {
+            throw new ArgumentNullException(nameof(policyName));
+        }
+
+        if (partitioner == null)
+        {
+            throw new ArgumentNullException(nameof(partitioner));
+        }
+
+        if (PolicyMap.ContainsKey(policyName) || UnactivatedPolicyMap.ContainsKey(policyName))
+        {
+            throw new ArgumentException("There already exists a policy with the name {name}");
+        }
```

Comment:
Why doesn't this override? Is this what the other policy based subsystems do?

Reply:
AuthZ AddPolicy overrides on subsequent calls, AuthN schemes throw on a duplicate scheme being added.

Reply:
CORS overrides as well, though to me it feels right that the first setting would win & that we'd inform users when they'd provided a duplicate - I'm open to persuasion, though